### PR TITLE
fix(frontend): stop reporting client errors as 500 on the vLLM chat-processor path

### PR DESCRIPTION
**Pull request by sungsooha**
## Summary

On `--dyn-chat-processor vllm` (the Python chat processor), two request-rejection paths surfaced as **HTTP 500 with no usable explanation**. The Rust `OpenAIPreprocessor` handles both correctly; the Python path does not go through it, so it needs its own handling.

Found while re-testing NVBug 6678456 against a current container. The ticket read as "valid parameters fail with 500", but the backend was rejecting these requests *correctly* — only the status and the message were being thrown away in the frontend.

## What was wrong

**1. A worker-side rejection lost its status.** An unsupported sampling parameter or an unparseable grammar crosses the Rust boundary as a plain exception whose text is:

```
BackendInvalidArgument: {"message":"The min_p and logit_bias sampling parameters are not yet
                         supported with speculative decoding.","code":400}
```

The payload already says `"code": 400`. The generic `except Exception` turned it into an internal error and dropped both. `backend_invalid_argument_to_http_error()` recovers them, mirroring the status rules of `extract_backend_error_if_present` (`lib/llm/src/http/service/openai.rs`) rather than inventing new ones.

**2. `logprobs`/`top_logprobs` accepted work it cannot complete.** Not implemented on this path (there is an existing `TODO`), but the code warned and continued, emitting a response with an empty logprobs struct that the HTTP layer cannot deserialize:

```
invalid length 0, expected struct ChatChoiceLogprobs with 2 elements
```

It now rejects up front.

## Why 400 and not 501

501 is the obvious choice — it is what this API uses for `previous_response_id`, `background`, and `prompt`, and matches the documented rule *"handlers return explicit 501 responses instead of accepting work that cannot complete yet."*

**It does not work from Python, and fails silently.** Those siblings are raised *inside* the Rust HTTP service. A status asserted by a Python engine is triaged by `SanitizedError::for_backend_status` (`lib/llm/src/http/service/error.rs`), which forwards a 4xx verbatim but **replaces the body of any 5xx**. Observed, not assumed:

```
HttpError: HTTP 501: Validation: `logprobs` ... are not supported
→ (500, {message: "Internal server error", details: {"backend_status": 501}})
```

So on this seam only a 4xx can carry an explanation to the client. That reasoning is in a comment at the call site so it is not "corrected" back to 501.

## Measured

Inkling-NVFP4 / GB300, 2-replica AGG, vLLM 0.28.0:

| case | before | after |
|---|---|---|
| `min_p=0.05` | 500 "Internal server error" | **400** "The min_p and logit_bias sampling parameters are not yet supported with speculative decoding." |
| malformed `json_schema` | 500 "Internal server error" | **400** "Grammar error: Invalid type: invalid_type" |
| `logprobs=true` | 500 "Internal server error" | **400** naming the unsupported field |
| `top_logprobs=3` | 500 "Internal server error" | **400** |

The other seven cases in the probe (`temperature`, `include_reasoning`, `guided_json`, `/v1/completions`, `/v1/responses`, `previous_response_id`, …) are **unchanged** — the mapping only converts errors carrying the `BackendInvalidArgument` discriminator, and the logprobs guard only fires when logprobs were requested.

Note `min_p` stays a **400, not a 200**: `sampling_params.py` rejects `min_p`/`logit_bias` with *any* speculative decoding, and this deployment runs MTP. The fix makes the failure honest; it does not add support.

## Tests

- `backend_invalid_argument_to_http_error` — 14 cases, including both real strings captured from a live frontend traceback, an outer-wrapped prefix, a non-JSON payload, missing/boolean/out-of-range codes, and four negatives that must fall through untouched.
- Two integration tests driving `_generate_and_stream` through a rejecting fake engine: one asserts the 400 and message survive, one asserts a plain `RuntimeError` still yields `internal_error` so the mapping stays narrow.
- `test_generator_preserves_zero_top_logprobs` pinned the warn-and-continue behaviour, so it is repointed at the rejection and renamed. It still guards the subtlety it always did: `logprobs=True` with `top_logprobs=0` projects to `0`, so a **truthiness** guard would let exactly that request through and re-create the 500 — the guard tests `is not None`.

`sglang_processor.py` has the identical generic handler and very likely the same defect. Left untouched here since it is unverified on that path; the helper lives in shared `frontend/utils.py` so it can adopt it.

## Summary by CodeRabbit

* **Bug Fixes**
  * Invalid sampling parameters now return clear HTTP 400 errors.
  * Unsupported log-probability requests are rejected instead of continuing with a warning.
  * Streaming backend validation errors now preserve their status codes and decoded messages.
  * Unrelated engine failures continue to return internal-error responses.
* **Tests**
  * Expanded coverage for structured, malformed, and unexpected backend errors, including status-code handling and edge cases.

### CODE REVIEW

**Review by rmccorm4 — DISMISSED** [review]

File: components/src/dynamo/frontend/tests/test_frontend_utils.py [hunk 1]
```diff
@@ -85,3 +86,73 @@ def test_none_response(self):
...
+        assert err is not None
+        assert err.code == 400 and err.message == "grammar is not valid"
+
+    def test_missing_code_defaults_to_400(self):
+        err = backend_invalid_argument_to_http_error(
+            ValueError('BackendInvalidArgument: {"message":"nope"}')
+        )
+        assert err is not None and err.code == 400
+
+    @pytest.mark.parametrize("code", [0, 99, 600, 1000, -1])
```
[the remark this answers is not in the document]
Comment (rmccorm4) [reply components/src/dynamo/frontend/tests/test_frontend_utils.py]: Addressed in [`6b4a4d2`](https://github.com/ai-dynamo/dynamo/pull/13980/commits/6b4a4d200ac6f74c6446fbd2145e240e8da5c9f6): the string-parser implementation and its parameterized tests were removed in favor of typed error propagation, so this redundant case table no longer exists.

File: components/src/dynamo/frontend/vllm_processor.py [hunk 2]
```diff
@@ -619,9 +621,26 @@ async def _generator_inner(
             sampling_params.logprobs = top_logprobs
         # TODO: Support logprobs in the distributed vLLM chat processor by
         # converting worker log_probs/top_logprobs into EngineCoreOutput.new_logprobs.
+        #
+        # Until then, REJECT rather than warn-and-continue. Proceeding emits a
```
[the remark this answers is not in the document]
Comment (rmccorm4) [reply components/src/dynamo/frontend/vllm_processor.py]: Addressed in [`7c97930`](https://github.com/ai-dynamo/dynamo/pull/13980/commits/7c97930f4213aecced25652c699170054b21ba04): unsupported logprobs are now rejected before preprocessing, and the verbose downstream narrative block was removed.

**Review by rmccorm4 — APPROVED** [review]
Approving to unblock for now.

Longer-term, I want to remove this backend string-prefix parsing entirely and make backend error propagation typed again (similar to https://github.com/ai-dynamo/dynamo/pull/13930).

Suggested follow-up direction:

- Stop treating `str(exc)` as a protocol; `BackendInvalidArgument: ...` should not be parsed out of arbitrary exception messages.
- Preserve typed backend errors only at the routed LLM/frontend boundary where we know the source and intended HTTP semantics.
- Avoid changing generic Dynamo runtime Python client behavior unless done deliberately: today those clients may rely on receiving plain `ValueError`s rather than Dynamo-specific exception types.
  - One option is to separate the routed-engine path from the existing generic response-stream path, so we can preserve typed routed errors without changing `AsyncResponseStream` behavior for other clients.

CC @biswapanda @GuanLuo
